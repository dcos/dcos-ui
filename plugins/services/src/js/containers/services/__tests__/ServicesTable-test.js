jest.dontMock("#SRC/js/components/CollapsingString");
jest.dontMock("../../../constants/HealthStatus");
jest.dontMock("../ServicesTable");
// Explicitly mock for react-intl
jest.mock("../../../components/modals/ServiceActionDisabledModal");

/* eslint-disable no-unused-vars */
const React = require("react");
/* eslint-enable no-unused-vars */
const ReactDOM = require("react-dom");
const ServicesTable = require("../ServicesTable");
const Application = require("../../../structs/Application");

describe("ServicesTable", function() {
  const healthyService = new Application({
    healthChecks: [{ path: "", protocol: "HTTP" }],
    cpus: 1,
    instances: 1,
    mem: 2048,
    disk: 0,
    tasksStaged: 0,
    tasksRunning: 2,
    tasksHealthy: 2,
    tasksUnhealthy: 0
  });

  beforeEach(function() {
    this.container = global.document.createElement("div");
    this.instance = ReactDOM.render(
      <ServicesTable.WrappedComponent />,
      this.container
    );
  });

  afterEach(function() {
    ReactDOM.unmountComponentAtNode(this.container);
  });

  describe("#renderStats", function() {
    it("should render resources/stats cpus property", function() {
      var cpusCell = ReactDOM.render(
        this.instance.renderStats("cpus", healthyService),
        this.container
      );

      expect(ReactDOM.findDOMNode(cpusCell).textContent).toEqual("1");
    });

    it("should render resources/stats mem property", function() {
      var memCell = ReactDOM.render(
        this.instance.renderStats("mem", healthyService),
        this.container
      );

      expect(ReactDOM.findDOMNode(memCell).textContent).toEqual("2 GiB");
    });

    it("should render resources/stats disk property", function() {
      var disksCell = ReactDOM.render(
        this.instance.renderStats("disk", healthyService),
        this.container
      );

      expect(ReactDOM.findDOMNode(disksCell).textContent).toEqual("0 B");
    });

    it("should render sum of resources/stats cpus property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0
      });

      var cpusCell = ReactDOM.render(
        this.instance.renderStats("cpus", application),
        this.container
      );

      expect(ReactDOM.findDOMNode(cpusCell).textContent).toEqual("2");
    });

    it("should render sum of resources/stats mem property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0
      });

      var memCell = ReactDOM.render(
        this.instance.renderStats("mem", application),
        this.container
      );

      expect(ReactDOM.findDOMNode(memCell).textContent).toEqual("4 GiB");
    });

    it("should render sum of resources/stats disk property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0
      });

      var disksCell = ReactDOM.render(
        this.instance.renderStats("disk", application),
        this.container
      );

      expect(ReactDOM.findDOMNode(disksCell).textContent).toEqual("0 B");
    });
  });
});
