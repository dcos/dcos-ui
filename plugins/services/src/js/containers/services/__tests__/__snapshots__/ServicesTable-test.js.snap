// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesTable #renderRegions renders with multiple regions 1`] = `
<span
  className="tooltip-wrapper tooltip-block-wrapper text-overflow"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  aws/eu-central-1, dc-east
</span>
`;

exports[`ServicesTable #renderRegions renders with no regions 1`] = `
<span
  className="tooltip-wrapper tooltip-block-wrapper text-overflow"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  N/A
</span>
`;

exports[`ServicesTable #renderRegions renders with one region 1`] = `
<span
  className="tooltip-wrapper tooltip-block-wrapper text-overflow"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  aws/eu-central-1
</span>
`;

exports[`ServicesTable #renderStatus renders with deploying status 1`] = `
<div
  className="flex"
>
  <div
    className=" service-status-icon-wrapper"
  >
    <div
      className="tooltip-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <svg
        aria-label="system-yield icon"
        className="css-1depasv"
        height={16}
        preserveAspectRatio="xMinYMin meet"
        role="img"
        viewBox="0 0 16 16"
        width={16}
      >
        <use
          xlinkHref="#system-yield"
        />
      </svg>
    </div>
    <span
      className="status-bar-text"
    >
      Deploying
    </span>
  </div>
  <div
    className="service-status-progressbar-wrapper"
  >
    <div
      className="tooltip-wrapper text-align-center"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="status-bar flex-box status-bar--large staged"
      >
        <span
          className="bar success"
          style={
            Object {
              "width": "30%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesTable #renderStatus renders with not available status 1`] = `
<div
  className="flex"
>
  <div
    className=" service-status-icon-wrapper"
  >
    <div
      className="tooltip-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <svg
        aria-label="system-yield icon"
        className="css-1depasv"
        height={16}
        preserveAspectRatio="xMinYMin meet"
        role="img"
        viewBox="0 0 16 16"
        width={16}
      >
        <use
          xlinkHref="#system-yield"
        />
      </svg>
    </div>
  </div>
  <div
    className="service-status-progressbar-wrapper"
  >
    <div
      className="tooltip-wrapper text-align-center"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="status-bar flex-box status-bar--large staged"
      >
        <span
          className="bar success"
          style={
            Object {
              "width": "30%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesTable #renderStatus renders with running status 1`] = `
<div
  className="flex"
>
  <div
    className="running-state service-status-icon-wrapper"
  >
    <div
      className="tooltip-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <svg
        aria-label="system-yield icon"
        className="css-1depasv"
        height={16}
        preserveAspectRatio="xMinYMin meet"
        role="img"
        viewBox="0 0 16 16"
        width={16}
      >
        <use
          xlinkHref="#system-yield"
        />
      </svg>
    </div>
    <span
      className="status-bar-text"
    >
      Running
    </span>
  </div>
  <div
    className="service-status-progressbar-wrapper"
  >
    <div
      className="tooltip-wrapper text-align-center"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="status-bar flex-box status-bar--large staged"
      >
        <span
          className="bar success"
          style={
            Object {
              "width": "30%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesTable #renderStatus renders with stopped status 1`] = `
<div
  className="flex"
>
  <div
    className=" service-status-icon-wrapper"
  >
    <div
      className="tooltip-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <svg
        aria-label="system-yield icon"
        className="css-1depasv"
        height={16}
        preserveAspectRatio="xMinYMin meet"
        role="img"
        viewBox="0 0 16 16"
        width={16}
      >
        <use
          xlinkHref="#system-yield"
        />
      </svg>
    </div>
    <span
      className="status-bar-text"
    >
      Stopped
    </span>
  </div>
  <div
    className="service-status-progressbar-wrapper"
  >
    <div
      className="tooltip-wrapper text-align-center"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="status-bar flex-box status-bar--large staged"
      >
        <span
          className="bar success"
          style={
            Object {
              "width": "30%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
